src/Support/Field.php

```diff
@@ -198,24 +200,53 @@ protected function getResolver(): ?Closure
                 }
             }
 
-            // Add the 'selects and relations' feature as 5th arg
-            if (isset($arguments[3])) {
-                $arguments[] = function (int $depth = null) use ($arguments): SelectFields {
-                    $ctx = $arguments[2] ?? null;
-
-                    return new SelectFields($arguments[3], $this->type(), $arguments[1], $depth ?? 5, $ctx);
-                };
-            }
-
             // Authorize
             if (call_user_func_array($authorize, $arguments) != true) {
                 throw new AuthorizationError('Unauthorized');
             }
 
-            return call_user_func_array($resolver, $arguments);
+            $method = new ReflectionMethod($this, 'resolve');
+
+            $additionalParams = array_slice($method->getParameters(), 3);
+
+            $additionalArguments = array_map(function ($param) use ($arguments) {
+                if ($param->getType()->getName() === 'Closure') {
```

Comment:
Seeing that the other uses reference FQ class names, this should be `Closure::class` (and be imported) I guess?